staging/src/k8s.io/apimachinery/pkg/runtime/serializer/json/json_test.go

```diff
@@ -124,13 +124,290 @@ func TestDecode(t *testing.T) {
 		errFn          func(error) bool
 		expectedObject runtime.Object
 		expectedGVK    *schema.GroupVersionKind
-	}{
+	}
+
+	testCases := []testCase{
+		// missing metadata without into, typed creater
+		{
+			data: []byte("{}"),
+
+			expectedGVK: &schema.GroupVersionKind{},
+			errFn:       func(err error) bool { return strings.Contains(err.Error(), "Object 'Kind' is missing in") },
```

Comment:
nit but, should we report both missing kind and version at the same time?

Reply:
not changing existing non-strict behavior in this PR